spec/memery_spec.rb

```diff
@@ -366,6 +393,47 @@ def self.macro(name)
     end
   end
 
+  describe "with multiple methods" do
+    let(:h) { H.new }
+
+    specify do
+      values = [h.m, h.n, h.m, h.n]
+      expect(values).to eq([:m, :n, :m, :n])
+      expect(CALLS).to eq([:m, :n])
+    end
+
+    specify do
+      values = [h.x, h.y, h.x, h.y]
+      expect(values).to eq([:x, :y, :x, :y])
+      expect(CALLS).to eq([:x, :y])
+
+      allow(Process).to receive(:clock_gettime).with(Process::CLOCK_MONOTONIC)
```

Comment:
Do we really need to test this stuff here? It looks like it's already tested above 🤔

Reply:
Plausibly not - I was trying to make sure I tested all of the behaviors you tested in the singleton case against the multiple-memoize call, but since it is actually exercising the same code inside the loop, it might be overkill. Happy to drop this

Reply:
oh, I remember what I was doing here. The `ttl` option needs to be passed into all the in-loop `define_memoized_method`, and this was the only way I saw of exercising that. Still happy to drop it if you prefer, I tend to err on the side of over-testing out of habit :-)

Reply:
I guess we can just collect everything with `**options` and pass them to `define_memoized_method`.

Reply:
You'd lose the parameter-checking. Currently calling `memoize :foo, baz: nil` gives a useful error indicating that _that's not a useful option_ - after such a change, they'd get the same error, but about a method they're not calling themselves. Not a crushing cost, but it doesn't seem like it buys us anything - I'd probably suggest it if you had a couple more parameters (or possibly extracting a parameters object), but I think it'd be overkill for two.